File: FRONTEND/backgammon-frontend/src/app/features/tournaments/models/api/responses/tournament-base.response.ts
import { TournamentUserState } from '../../enums/tournament-user-state.type';
import { RulesTemplateResponse } from '../../../../rules-template/models/api/responses/rules-template.response';

export interface TournamentBaseResponse {
  id: string;

  name: string;
  description?: string;

  type: string;
  visibility: string;
  status: string;

  maxParticipants: number;

  startDate: string;
  endDate: string;
  deadline: string;

  organizerUserName: string;

  rulesTemplate: RulesTemplateResponse;

  tournamentUserState: TournamentUserState | null;
}

File: BACKEND/Application/Tournaments/Responses/TournamentBaseResponse.cs
using Application.RulesTemplate.Responses;

namespace Application.Tournaments.Responses
{
    public class TournamentBaseResponse
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public required string Type { get; set; }
        public required string Visibility { get; set; }
        public required string Status { get; set; }
        public int MaxParticipants { get; set; }
        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset EndDate { get; set; }
        public DateTimeOffset Deadline { get; set; }
        public required string OrganizerUserName { get; set; }
        public required RulesTemplateResponse RulesTemplate { get; set; }
        public string? TournamentUserState { get; set; }
    }
}
